pkg/chain/ethereum/ethereum.go

```diff
@@ -237,7 +244,42 @@ func (ec *ethereumChain) SubmitRelayEntry(
 		}
 	}()
 
-	_, err = ec.keepRandomBeaconOperatorContract.RelayEntry(entry)
+	transactorOptions := bind.NewKeyedTransactor(ec.accountKey.PrivateKey)
+	from := transactorOptions.From
+
+	toAddress, err := ec.config.ContractAddress("KeepRandomBeaconOperator")
+	if err != nil {
+		logger.Errorf("failed to find KeepRandomBeaconOperator address [%v]", err)
+	}
+
+	abi, err := goethabi.JSON(strings.NewReader(keepcoreabi.KeepRandomBeaconOperatorABI))
+	if err != nil {
+		logger.Errorf("failed to instantiate ABI: [%v]", err)
+	}
+
+	packed, err := abi.Pack("relayEntry", entry)
+	if err != nil {
+		logger.Errorf("failed packing relayEntry to conform with ABI [%v]", err)
+	}
+
+	msg := ethereum.CallMsg{
+		From: from,
+		To:   toAddress,
+		Data: packed,
+	}
+
+	gasEstimate, err := ec.client.EstimateGas(context.TODO(), msg)
+	if err != nil {
+		logger.Errorf("failed to estimate gas for relay entry [%v]", err)
+	}
+	gasEstimateWithMargin := float64(gasEstimate) * float64(1.2) // 20% more than original
```

Comment:
`float64(1.2)` could be a `gasMultiplier` in `ethutil.TransactionOptions{`. If `gasMultiplier` is supplied, the contract executes `*EstimateGas` and calculates the final value. Then, we can reduce this code to just:
```
_, err = ec.keepRandomBeaconOperatorContract.RelayEntry(
  entry,
  ethutil.TransactionOptions{
    GasMultiplier: float64(1.2),
  },
)
```

What do you think?

Reply:
agree, but we'd also need to pass `GasLimit` along with `GasMultiplier` either case:

```
_, err = ec.keepRandomBeaconOperatorContract.RelayEntry(
  entry,
  ethutil.TransactionOptions{
    GasLimit: 1234,
    GasMultiplier: float64(1.2),
  },
)
```

Reply:
Why the gas limit? For relay entry we can't set a gas limit because of a callback.

Reply:
I mean the gas estimated earlier that would be assigned to `GasLimit` just like in this PR.

From what I see if we don't provide `GasLimit` in TransactionOptions then we estimate gas downstream in `go-ethereum/accounts/abi/bind/base.go::L222` before sending transaction. Unless I miss smth, then passing just `GasMultiplier` without `GasLimit` won't work.

Reply:
If `GasMultiplier` is provided, the generated contract code need to do the gas estimation part, that's my understanding. We can do easier option first, though. Expose `*GasEstimate` functions and instead of `GasMultipler`, call `RelayEntryGasEstimate` here, multiply manually, and pass the resulting value as `GasLimit`.

Reply:
> If GasMultiplier is provided, the generated contract code need to do the gas estimation part, that's my understanding.

Ah.. I see, then in ex. `options.go::Apply()` we would need to call the estimation function and set `GasLimit` multiplied by ex. `1.2`? 

> We can do easier option first, though.
K, I'll go with that first.

Reply:
@pdyraga ready for another look